File: mobile/src/main/java/info/papdt/express/helper/EntryActivity.java
package info.papdt.express.helper;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatDelegate;

import info.papdt.express.helper.support.Settings;
import info.papdt.express.helper.ui.MainActivity;
import info.papdt.express.helper.ui.SplashActivity;

public class EntryActivity extends Activity {

	@Override
	public void onCreate(Bundle savedInstanceState) {
		Settings mSettings = Settings.getInstance(getApplicationContext());

		super.onCreate(savedInstanceState);

		/** Open activity */
		if (mSettings.getBoolean(Settings.KEY_FIRST_RUN, true)) {
			mSettings.putBoolean(Settings.KEY_FIRST_RUN, false);
			Intent intent = new Intent(this, SplashActivity.class);
			intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
			startActivity(intent);
		} else {
			MainActivity.launch(this);
		}

		finish();
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/EntryActivity.kt
package info.papdt.express.helper

import android.app.Activity
import android.content.Intent
import android.os.Bundle

import info.papdt.express.helper.support.SettingsInstance
import info.papdt.express.helper.ui.HomeActivity
import info.papdt.express.helper.ui.SplashActivity

class EntryActivity : Activity() {

	public override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)

		/** Open activity  */
		if (SettingsInstance.firstRun) {
			SettingsInstance.firstRun = false
			val intent = Intent(this, SplashActivity::class.java)
			intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
			startActivity(intent)
		} else {
			// MainActivity.launch(this)
			startActivity(Intent(this, HomeActivity::class.java)
                    .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK))
		}

		finish()
	}

}
